ui/app/services/auth.js

```diff
@@ -95,18 +99,15 @@ export default Service.extend({
     return this.ajax(url, 'POST', { namespace });
   },
 
-  calculateExpiration(resp, creationTime) {
-    const creationTTL = resp.creation_ttl || resp.lease_duration;
-    const leaseMilli = creationTTL ? creationTTL * 1e3 : null;
-    const tokenIssueEpoch = resp.creation_time ? resp.creation_time * 1e3 : creationTime || Date.now();
-    const tokenExpirationEpoch = tokenIssueEpoch + leaseMilli;
-    const expirationData = {
-      tokenIssueEpoch,
+  calculateExpiration(resp) {
+    let now = this.now();
+    const ttl = resp.ttl || resp.lease_duration;
+    const tokenExpirationEpoch = now + ttl * 1e3;
+    this.set('expirationCalcTS', now);
+    return {
+      ttl,
       tokenExpirationEpoch,
-      leaseMilli,
     };
-    this.set('expirationCalcTS', Date.now());
```

Comment:
We don't set this anymore now? Do we just calculate the expiration on the fly every time now?

Reply:
Still there, just got moved up: https://github.com/hashicorp/vault/pull/5435/files/d153810d3491a11942fcd1fc62ee7dd97845d0ff#diff-10502f4cc8e730c6150cac0331e18b9bR106

Reply:
aha!